packages/create-react-admin/src/generateProject.ts

```diff
@@ -0,0 +1,193 @@
+import path from 'path';
+import fs from 'fs';
+import fsExtra from 'fs-extra';
+import { ProjectConfiguration } from './ProjectState.js';
+
+export const generateProject = async (state: ProjectConfiguration) => {
+    const projectDirectory = initializeProjectDirectory(state.name);
+
+    copyDirectoryFiles(
+        path.join(__dirname, '../templates/common'),
+        projectDirectory
+    );
+
+    if (state.dataProvider !== 'none') {
+        copyDirectoryFiles(
+            path.join(__dirname, '../templates', state.dataProvider),
+            path.join(path.join(projectDirectory, 'src'))
+        );
+    }
+
+    if (state.authProvider !== 'none') {
+        copyDirectoryFiles(
+            path.join(__dirname, '../templates', state.authProvider),
+            path.join(path.join(projectDirectory, 'src'))
+        );
+    }
+
+    generateAppFile(projectDirectory, state);
+    generatePackageJson(projectDirectory, state);
+    generateEnvFile(projectDirectory, state);
+};
+
+const generateAppFile = (
+    projectDirectory: string,
+    state: ProjectConfiguration
+) => {
+    fs.writeFileSync(
+        path.join(projectDirectory, 'src', 'App.tsx'),
+        `
+import { Admin, Resource, ListGuesser, EditGuesser, ShowGuesser } from 'react-admin';
+${
+    state.dataProvider !== 'none'
+        ? `import { dataProvider } from './dataProvider';\n`
+        : ''
+}${
+            state.authProvider !== 'none'
+                ? `import { authProvider } from './authProvider';\n`
+                : ''
+        }
+export const App = () => (
+    <Admin
+        ${
+            state.dataProvider !== 'none'
+                ? `dataProvider={dataProvider}\n\t\t`
+                : ''
+        }${
+            state.authProvider !== 'none' ? `authProvider={authProvider}\n` : ''
+        }>
```

Comment:
This closing `>` is not well indented

Reply:
Fixed

Reply:
No it's still off

![2023-04-24_14-18](https://user-images.githubusercontent.com/14542336/233994152-c0796932-dc9c-4b0a-b9bd-195475a45af5.png)